packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -153,6 +155,38 @@ export function convertWireToOsdkObjects<
   objs: OntologyObjectV2[],
 ) {
   for (const obj of objs) {
+    if (obj.__rid) {
+      obj.$rid = obj.__rid;
+      delete obj.__rid;
+    }
+
+    // Backend returns as __apiName but we want to stick to $ structure
+    obj.$apiName = obj.__apiName;
+
+    // for now these are the same but when we start doing interface projections the $objectType will always be underlying and
+    // the $apiName will be for the current view (in current designs)
+    obj.$objectType = obj.__apiName;
+
+    // copying over for now as its always returned. In the future, this should just be inferred from underlying
+    obj.$primaryKey = obj.__primaryKey;
+
+    // After Feb 2024 (unless we have new apis):
+    if (isAfterFeb2024OrNewApis) {
+      delete obj.__apiName;
+      delete obj.__primaryKey;
+    } else {
+      // Hide these from things like `console.log` so that people
+      // don't think to use them.
+      Object.defineProperties(obj, {
+        "__apiName": {
+          enumerable: false,
+        },
+        "__primaryKey": {
+          enumerable: false,
+        },
+      });
```

Comment:
So its a little weird that we're afraid about conflicts here given that the backend is sending us these over the wire, they would've had to avoid this conflict ahead of time.

Additionally, I think there's a bunch more references to __apiName and __primaryKey in client that need to get worked through. Check out `OsdkObject.ts` and `OsdkObjectFrom.ts` for the root of that typing

Reply:
Re conflicts, I am looking at consistency. In 2.0, the things that are "OSDK" on the objects have `$` names and the things that dont, dont. This is more about consistency than conflicts. Correctness is a benefit.